src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -259,7 +257,7 @@ public boolean upload(FilePath workspace, String artifactIncludes, TaskListener
                 SlackFileRequest slackFileRequest = new SlackFileRequest(
                 workspace, populatedToken, roomId, null, artifactIncludes);
                 try {
-                    workspace.getChannel().callAsync(new SlackUploadFileRunner(log, Jenkins.get().proxy, slackFileRequest)).get();
+                    workspace.getChannel().callAsync(new SlackUploadFileRunner(log, slackFileRequest)).get();
```

Comment:
everything of the call async is ran on the agent where any methods under `Jenkins.get()` and many others will fail. Any information about Jenkins needs to be passed into the constructor which was what this was doing.

So this change won't work, (and it is why it was done in this way)

Reply:
oh I didn't know that.
Thank you for explanation.